packages/twenty-server/src/graphql-config.service.ts

```diff
@@ -0,0 +1,130 @@
+import { exceptionFilters } from 'src/filters';
+import { ExceptionHandlerService } from 'src/integrations/exception-handler/exception-handler.service';
+
+@Injectable()
+export class GraphQLConfigService
+  implements GqlOptionsFactory<YogaDriverConfig<'express'>>
+{
+  constructor(
+    private readonly tokenService: TokenService,
+    private readonly exceptionHandlerService: ExceptionHandlerService,
+    private readonly moduleRef: ModuleRef,
+  ) {}
+
+  createGqlOptions(): YogaDriverConfig {
+    return {
+      context: ({ req }) => ({ req }),
+      autoSchemaFile: true,
+      include: [CoreModule],
+      conditionalSchema: async (context) => {
+        try {
+          let workspace: Workspace;
+
+          // If token is not valid, it will return an empty schema
+          try {
+            workspace = await this.tokenService.validateToken(context.req);
+          } catch (err) {
+            return new GraphQLSchema({});
+          }
+
+          return await this.createSchema(context, workspace);
+        } catch (error) {
+          // Filter the exception
+          this.filterException(error);
+
+          // Send the unfiltered error to Sentry
+          this.exceptionHandlerService.captureException(error);
+
+          throw error;
+        }
+      },
+      resolvers: { JSON: GraphQLJSON },
+      plugins: [],
+    };
+  }
+
+  async createSchema(
+    context: YogaDriverServerContext<'express'> & YogaInitialContext,
+    workspace: Workspace,
+  ): Promise<GraphQLSchemaWithContext<YogaDriverServerContext<'express'>>> {
+    // Create a new contextId for each request
+    const contextId = ContextIdFactory.create();
+
+    // Register the request in the contextId
+    this.moduleRef.registerRequestByContextId(context.req, contextId);
+
+    // Resolve the WorkspaceFactory for the contextId
+    const workspaceFactory = await this.moduleRef.resolve(
+      WorkspaceFactory,
+      contextId,
+      {
+        strict: false,
+      },
+    );
+
+    return await workspaceFactory.createGraphQLSchema(workspace.id);
+  }
+
+  private filterException(error: unknown) {
+    if (error instanceof JsonWebTokenError) {
+      //mockedUserJWT
+      throw new GraphQLError('Unauthenticated', {
+        extensions: {
+          code: 'UNAUTHENTICATED',
+        },
+      });
+    }
+
+    if (error instanceof TokenExpiredError) {
+      throw new GraphQLError('Unauthenticated', {
+        extensions: {
+          code: 'UNAUTHENTICATED',
+        },
+      });
+    }
+
+    /**
+     * Not really pretty but it works for now as we're outside of Nest.JS "flow" `APP_FILTER` is not triggered
+     * Loop through all the exception filters and check if the error is instance of any of the metadata
+     */
+    for (const exceptionFilter of exceptionFilters) {
+      // Extract the metadata from the exception filter
+      const metadata: Array<Type<any> | Abstract<any>> | undefined =
```

Comment:
It works but I don't really see the point of doing that :
Why not:
- define http-exception-handler.util.ts 
- define typeorm-exception-handler.util.ts
- define global-exception-handler.util.ts which is detecting if we should use http-exception-handler.util or typeorm-exception-handler.util based on Exception type + catching all other exceptions

Only register global-exception.filter that will call global-exception-handler.util

Then in graphql-config.service, we can also call global-exception-handler.util.ts without hacking into the nestjs filter API.